resources/session04/mashup_6.py

```diff
@@ -0,0 +1,263 @@
+        (u'Total Inspections', samples)
+    ]
+    return data
+
+
+def result_generator(count):
+    use_params = {
+        'Inspection_Start': '01/26/2015',
+        'Inspection_End': '201/26/2016',
+        'Zip_Code': '98101'
+    }
+    # html = get_inspection_page(**use_params)
+    sorting_key, marker_type = check_sorting()
+    html = load_inspection_page('inspection_page.html')
+    parsed = parse_source(html)
+    content_col = parsed.find("td", id="contentcol")
+    data_list = restaurant_data_generator(content_col)
+    for data_div in data_list[:count]:
+        metadata = extract_restaurant_metadata(data_div)
+        inspection_data = get_score_data(data_div)
+        metadata.update(inspection_data)
+        # sorting_value = str(metadata.get(sorting_key))
+        yield metadata
+
+
+def create_ordered_dict_and_sort(dictionary, key=None):
+    dictionary = OrderedDict(dictionary)
+    if key:
+        if key in dictionary:
+            dictionary.move_to_end(key, last=False)
+            return dictionary
+    else:
+        return dictionary
+
+
+def get_geojson(result):
+    address = " ".join(result.get('Address', ''))
+    if not address:
+        return None
+    geocoded = geocoder.google(address)
+    geojson = geocoded.geojson
+    inspection_data = {}
+    use_keys = (
+        'Business Name', 'Average Score', 'Total Inspections', 'High Score'
+    )
+    for key, val in result.items():
+        if key not in use_keys:
+            continue
+        if isinstance(val, list):
+            val = " ".join(val)
+        inspection_data[key] = val
+    sorting_key, marker_type = check_sorting()
+    marker_value = inspection_data.get(sorting_key)
+    if marker_type == 'graduated':
+            inspection_data['marker-color'] = get_color_graduated(marker_value)
+    if marker_type == 'shaded':
+            inspection_data['marker-color'] = get_color_shaded(marker_value)
+    # inspection_data = create_ordered_dict_and_sort(inspection_data, sorting_key)
+    geojson['sort_by'] = marker_value
+    geojson['properties'] = inspection_data
+    # print(geojson)
+    return geojson
+
+
+def get_color_graduated(score):
+    if score >= 90:
+        return '#006400'
+    elif score >= 80:
+        return '#9acd32'
+    elif score >= 70:
+        return '#ffff00'
+    elif score >= 60:
+        return '#ffd700'
+    elif score >= 50:
+        return '#ffa500'
+    elif score >= 40:
+        return '#ff4500'
+    elif score >= 30:
+        return '#ff0000'
+    else:
+        return '#8b0000'
+
+
+def get_color_shaded(score):
+    if score >= 10:
+        return '#000066'
+    elif score >= 8:
+        return '#0000b3'
+    elif score >= 6:
+        return '#0000ff'
+    elif score >= 4:
+        return '#4d4dff'
+    elif score >= 2:
+        return '#b3b3ff'
+    elif score == 1:
+        return '#e5e5ff'
+    else:
+        return '#666699'
+
+
+def check_sorting():
+    marker_type = None
+    if len(argv) >= 2:
```

Comment:
it is a little dangerous to rely on `argv` as imported from sys in more than one place.  It's a global value, but it is mutable, which means any of your other functions (or some other tool you are using) might change it from what you expect it to be.  It's best to isolate the places you touch sys.argv to the absolute minimum.